File: java/src/blog/ds/StackLinkedList.java
package blog.ds;

import java.util.StringJoiner;

public class StackLinkedList {
    Node head = null;

    public String toString() {
        StringJoiner joiner = new StringJoiner(",");
        Node tmp = this.head;

        while(tmp != null) {
            joiner.add(Integer.toString(tmp.data));
            tmp = tmp.next;
        }

        return joiner.toString();
    }

    // addFirst
    public void push(int data) {
        Node node = new Node(data);

        node.next = this.head;
        this.head = node;
    }

    // removeFirst
    public Integer pop() {
        if (this.empty()) {
            return null;
        } else {
            Node delete = this.head;
            int result = delete.data;
            Node newHead = delete.next;

            this.head = newHead;

            delete = null;
            return result;
        }
    }

    public boolean empty() {
        return this.head == null;
    }

    public Integer peek() {
        if (this.empty())
            return null;
        return this.head.data;
    }

    public int search(int data) {
        Node tmp = this.head;
        int idx = 0;

        while(tmp != null) {
            if (data == tmp.data)
                return idx;
            idx++;
            tmp = tmp.next;
        }
        return -1;
    }
}

File: python/blog/ds/stackLinkedList.py
from node import Node

class StackLinkedList:
    head = None

    def toString(self):
        result = []
        tmp = self.head
        while (tmp is not None):
            result.append(str(tmp.data))
            tmp = tmp.next

        return ','.join(result)

    # addFirst
    def push(self, data):
        node = Node(data)

        node.next = self.head
        self.head = node

    # removeFirst
    def pop(self):
        if self.empty():
            return None
        else:
            delete = self.head
            result = delete.data
            newHead = delete.next

            self.head = newHead

            delete = None
            return result

    def empty(self):
        return self.head is None;

    def peek(self):
        if self.empty():
            return None
        return self.head.data

    def search(self, data):
        tmp = self.head
        idx = 0

        while (tmp is not None):
            if data == tmp.data:
                return idx
            tmp = tmp.next
            idx += 1

        return -1
